### ISSUE TITLE
wasmplugin doesn't support for SIDECAR_OUTBOUND

### ISSUE THREAD

**Issue by Adol1111** [istio/istio#36238]
### Bug Description

Test WasmPlugin in 1.12.0
```yaml
apiVersion: extensions.istio.io/v1alpha1
kind: WasmPlugin
metadata:
  name: custom-filter
  namespace: default
spec:
  url: oci://harbor-host/test/custom-filter:0.1
```

```bash
$ istioctl pc listener demo-78cfb7f5cf-zj4dz -o yaml
```

All custom-filter found in inbound filters,  no outbound filters
```
httpFilters:
- configDiscovery:
     configSource:
       ads: {}
       initialFetchTimeout: 0s
       resourceApiVersion: V3
    typeUrls:
    - type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
  name: default.custom-filter
....
routeConfig:
  name: inbound|2222||
```

---
Then I use envoyfilter to append a wasm filter

```yaml
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: custom-filter
spec:
  configPatches:
  - applyTo: HTTP_FILTER
    match:
      context: ANY
      listener:
        portNumber: 2222
    patch:
      operation: INSERT_BEFORE
      value:
        name: custom-filter
        config_discovery:
          config_source:
            ads: {}
            initial_fetch_timeout: 0s # wait indefinitely to prevent filter chain being disabled
          type_urls: [ "type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm"]
---
apiVersion: networking.istio.io/v1alpha3
kind: EnvoyFilter
metadata:
  name: custom-filter-config
spec:
  configPatches:
  - applyTo: EXTENSION_CONFIG
    match:
      context: ANY
    patch:
      operation: ADD
      value:
        name: custom-filter
        typed_config:
          "@type": type.googleapis.com/udpa.type.v1.TypedStruct
          type_url: type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
          value:
            config:
              vm_config:
                vm_id: custom-filter_vm
                runtime: envoy.wasm.runtime.v8
                code:
                  remote:
                    http_uri:
                      uri: http://test-host/filter.wasm
                      timeout: 10s
              configuration:
                '@type': type.googleapis.com/google.protobuf.StringValue
                value: |
                  {}
```

outbound
``` yaml
....
- accessLog:
      ....
  address:
    socketAddress:
      address: 0.0.0.0
      portValue: 2222
  filterChains:
  - filterChainMatch:
       ...
    filters:
    - name: envoy.filters.network.http_connection_manager
      typedConfig:
        '@type': type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
      httpFilters:
        - configDiscovery:
            configSource:
              ads: {}
              initialFetchTimeout: 0s
            typeUrls:
            - type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
          name: custom-filter
       ....
  name: 0.0.0.0_2222
  trafficDirection: OUTBOUND
```

inbound
```yaml
httpFilters:
- configDiscovery:
    configSource:
      ads: {}
      initialFetchTimeout: 0s
    typeUrls:
    - type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
  name: custom-filter
....
routeConfig:
    name: inbound|2222||
    validateClusters: false
```

I see test test case replace envoyfilter with wasmplugin. But old test case match context is any. It is break now?

https://github.com/istio/istio/pull/33374/files#diff-ede888690c439e307a114381c85c1c7f0593876316691cd741d535317c9d3a34L1 

tests/integration/telemetry/stats/prometheus/wasm/testdata/bad-filter.yaml

can wasmplugin support match like envoyfilter？ 

### Version

```prose
$ istioctl version
client version: 1.12.0
control plane version: 1.12.0
data plane version: 1.12.0 (3 proxies)
$ kubectl version --short
Client Version: v1.18.8
Server Version: v1.18.8
```

**Comment by bianpengyuan** [comment]
This is a known limitation that Wasm plugin does not support outbound at the moment. cc @dgn @mandarjog

**Comment by Adol1111** [comment]
Will support outbound  in the futre?

**Comment by bianpengyuan** [comment]
> Will support outbound in the futre?

Yes.

**Comment by bianpengyuan** [comment]
not stale.

**Comment by wxdao** [comment]
@bianpengyuan Would you please clarify the limitation in the docs? It'll help other people not to waste more time debugging on this...

**Comment by ingwonsong** [comment]
After https://github.com/istio/istio/pull/38595, Wasm modules seems to be injected to both of in/outbound.

**Comment by wxdao** [comment]
> After #38595, Wasm modules seems to be injected to both of in/outbound.

It's reverted later in #39154

**Comment by zirain** [comment]
wait for https://github.com/istio/api/pull/2412

**Comment by dgn** [comment]
@ingwonsong I suppose this issue was addressed by your PR #40862 ?

**Comment by ingwonsong** [comment]
@dgn Sorry for super late response. yes. this was resolved.
